src/order/category/Semilattice.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.category.PartialOrder
+import order.hom.lattice
+
+/-!
+# The categories of semilattices
+
+This defines `SemilatticeSup` and `SemilatticeInf`, the categories of sup-semilattices with a bottom
+element and inf-semilattices with a top element.
+
+## References
+
+* [nLab, *semilattice*](https://ncatlab.org/nlab/show/semilattice)
+-/
+
+universes u
+open category_theory
+
+/-- The category of sup-semilattices with a bottom element. -/
+structure SemilatticeSup : Type.{u + 1} :=
```

Comment:
```suggestion
structure SemilatticeSup : Type.{u+1} :=
```
I think the convention for universe addition is to omit the space (this is a different operator to normal addition!)